### ISSUE TITLE
Error after upgrading to 0.3 with nextjs

### ISSUE THREAD

**Issue by dylankilkenny** [EthWorks/useDApp#120]
My app is server side rendered and was functioning perfect on 0.2, but since 0.3 I receive this error on startup:

```
Server Error
ReferenceError: window is not defined
This error happened while generating the page. Any console logs will be displayed in the terminal window.

Call Stack
getItem
file:/Users/dylankilkenny/dev/github/pxlgen/node_modules/@usedapp/core/dist/src/hooks/useLocalStorage.js (6:16)
<unknown>
file:/Users/dylankilkenny/dev/github/pxlgen/node_modules/@usedapp/core/dist/src/hooks/useLocalStorage.js (29:52)
useReducer
file:/Users/dylankilkenny/dev/github/pxlgen/node_modules/react-dom/cjs/react-dom-server.node.development.js (1537:57)
Object.useState
file:/Users/dylankilkenny/dev/github/pxlgen/node_modules/react-dom/cjs/react-dom-server.node.development.js (1475:10)
Object.useState
file:/Users/dylankilkenny/dev/github/pxlgen/node_modules/react/cjs/react.development.js (1508:21)
Object.useLocalStorage
file:/Users/dylankilkenny/dev/github/pxlgen/node_modules/@usedapp/core/dist/src/hooks/useLocalStorage.js (29:22)
NotificationsProvider
file:/Users/dylankilkenny/dev/github/pxlgen/node_modules/@usedapp/core/dist/src/providers/notifications/provider.js (12:22)
```

**Comment by marekkirejczyk** [comment]
I am not sure if we can make useDApp server-side renderable at all. With core useDApp feature being refreshing stuff often, how would that work?

**Comment by nezouse** [comment]
Okay, so the issue here is that we try to access local storage in one of the providers. 
If it was only some component you could render it client-side only, but in this case we can't really do anything to help here as now.

**Comment by Sagato** [comment]
Yeah I checked the code, there is simply no null check if window exists. useLocalStorage is using getItem(key), which is retrieving the localStorage item to initialize. What I would do is maybe this:

`export function useLocalStorage(key: string) {
  const [value, setValue] = useState() // instead of () => getItem(key)

  useEffect(() => {
    setValue(getItem(key))
  }, [key])

  useEffect(() => {
    setItem(key, value)
  }, [value, key])

  return [value, setValue] as const
}`

Maybe this would help. AFAIK the useEffect is called after intial rendering which means the window Object should be there.
Maybe I am mistaken.

greets

**Comment by Sagato** [comment]
Okay this doesn't work as expected. I currently don't know how to fix that issue.

**Comment by nezouse** [comment]
@Sagato  it didn't, as getItem is used in the useEffect below. In next this code is executed twice, once on the server, and then in the browser.
I created a PR so it should be fixed soon.

**Comment by Sagato** [comment]
I did the same :D I am just an idiot and changed js in uncompiled code and was wondering why it doesnt take effect :D 
Thx, I just tested it and it works. Unfortunenately I am receiving another error now when testing on localhost with ganache:
**provider.js?28e5:30 Missing multicall address for chain id 1337**

**Comment by marekkirejczyk** [comment]
@nezouse Merged and published in 0.3.4 Thank you ❤️ 

@Sagato Yeah, for Ganache to fully work we will need to figure out how to deploy mulitcall to local env.
For now just connection and sending transactions will work.

**Comment by DakshMiglani** [comment]
> For now just connection and sending transactions will work.

Just tried it, even just calling `activateBrowserWallet` causes the error message to appear!
